src/utils/arrow.rs

```diff
@@ -54,12 +53,18 @@ pub(crate) fn arrow_schema_from_column_names_and_range(
     column_names: &[String],
     row_idx: usize,
 ) -> Result<Schema> {
-    let mut fields = Vec::with_capacity(column_names.len());
-
-    for (col_idx, name) in column_names.iter().enumerate() {
-        let col_type = get_arrow_column_type(range, row_idx, col_idx)?;
-        fields.push(Field::new(&alias_for_name(name, &fields), col_type, true));
-    }
+    let fields = column_names
+        .iter()
+        .enumerate()
+        .map(|(col_idx, name)| {
+            let col_type = get_arrow_column_type(range, row_idx, col_idx).unwrap();
+            Field::new(
+                alias_for_name(name, &Vec::with_capacity(column_names.len())),
+                col_type,
+                true,
+            )
+        })
+        .collect::<Vec<_>>();
```

Comment:
I don't see where we do re-allocation! Maybe I didn't sleep enough

Reply:
* `alias_for_name` receives a new vector of the size of `column_names` (`Vec::with_capacity`) so the're one alloc per iteration. And actually this is wrong, since this vector is supposed to contain existing fields. So here, the alias generation is borked 
* IIRC `collect::<Vec<_>>` will do `.push` under the hood since not all iterator sizes can be known.

Reply:
ahhhhh I didn't see the new vec in `alias_for_name` lmao. God I'm tired! Thanks